### PR TITLE
Minor cleanups on halo and indexing

### PR DESCRIPTION
**Pull request by naoyam**
This is a cleanup PR, mostly for the changes introduced by PR #1734.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/reference_tensor.h [hunk 1]
```diff
@@ -1,27 +0,0 @@
-#pragma once
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/reference_tensor.h:1]: No longer used

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 2]
```diff
@@ -790,15 +789,14 @@ bool IndexCompute::isZero(IterDomain* id) const {
 IndexCompute IndexCompute::updateIndexCompute(
     const TensorDomain* new_td,
     const std::unordered_map<IterDomain*, IterDomain*>& id_map,
-    const ContigIDs& contig_finder,
-    const std::unordered_map<IterDomain*, Val*>& reference_halo_extent_map)
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp:794]: `reference_halo_extent_map` was always an empty set

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 3]
```diff
@@ -827,25 +830,7 @@ IndexCompute IndexCompute::updateIndexCompute(
       updated_zero_merged_in,
       contig_finder,
       {},
-      reference_halo_extent_map);
-
-  if (concrete_id_pass_) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp:832]: concrete_id_pass_ was always true

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 4]
```diff
@@ -827,25 +830,7 @@ IndexCompute IndexCompute::updateIndexCompute(
...
-    //   ca mapping and in the concrete_id_pass case, the id_map
-    //   also represents exact ca mapping.
-    // TODO: might need to re-visit pathological cases when we may
-    //  need to traverse and propagate halo info again in here.
-    for (auto id_entry : id_map) {
-      IterDomain* prev_id = id_entry.first;
-      IterDomain* new_id = id_entry.second;
-      auto halo_extent_it = reference_halo_extent_map_.find(prev_id);
-      if (halo_extent_it != reference_halo_extent_map_.end()) {
-        updated_index_compute.reference_halo_extent_map_[new_id] =
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp:844]: What we want is a halo extent map for the new IterDomains, which can be created by propagating the mappings for the concrete domains.

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 5]
```diff
@@ -441,9 +441,8 @@ void IndexCompute::handle(Merge* merge) {
 
   // When the reference has halo extent for inner_id, that extent needs to
   // be used to un-merge
-  if (reference_halo_extent_map_.find(inner_id) !=
-      reference_halo_extent_map_.end()) {
-    inner_extent = reference_halo_extent_map_[inner_id];
+  if (halo_extent_map_.find(inner_id) != halo_extent_map_.end()) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp:444]: Just dropped `reference_` as it's not reference anymore

**Review by csarofeen — APPROVED** [review]
LGTM

**Pull request merged**